qiskit_experiments/framework/analysis_result.py

```diff
@@ -10,46 +10,48 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""Analysis result abstract interface."""
+"""Analysis result class."""
 
 import copy
 import logging
 import math
 import uuid
+import traceback
 from typing import Optional, List, Union, Dict, Any
 
 import uncertainties
+
+from qiskit_ibm_experiment import IBMExperimentService, AnalysisResultData
+from qiskit_ibm_experiment import ResultQuality
+
 from qiskit_experiments.framework.json import (
     ExperimentEncoder,
     ExperimentDecoder,
     _serialize_safe_float,
 )
 
-from .database_service import DatabaseServiceV1
-from .db_fitval import FitVal
-from .device_component import DeviceComponent, to_component
-from .exceptions import DbExperimentDataError
-from .utils import save_data, qiskit_version
+from qiskit_experiments.database_service.db_fitval import FitVal
+from qiskit_experiments.database_service.device_component import DeviceComponent, to_component
+from qiskit_experiments.database_service.exceptions import ExperimentDataError
+from qiskit_experiments.database_service.utils import qiskit_version
 
 LOG = logging.getLogger(__name__)
 
 
-class DbAnalysisResult:
-    """Base common type for all versioned DbAnalysisResult abstract classes.
-
-    Note this class should not be inherited from directly, it is intended
-    to be used for type checking. When implementing a custom DbAnalysisResult
-    you should use the versioned classes as the parent class and not this class
-    directly.
-    """
+class AnalysisResult:
+    """Class representing an analysis result for an experiment.
 
-    version = 0
+    Analysis results can also be stored in a database.
 
+    The field `db_data` is a dataclass (`ExperimentDataclass`) containing
+    all the data that can be stored in the database and loaded from it, and
+    as such is subject to strict conventions.
```

Comment:
Please detail the strict conventions, in particular that certain fields must be stored under `result_data`

Reply:
Done